### PR TITLE
docs: ui capabilities text area PTL-1580

### PR DESCRIPTION
**Pull request by Gareth-Spencer-Genesis**
ATENTION! YOU SHOULD NOW BRANCH FROM PREPROD WHEN YOU UPDATE 

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
No

Have you done a trial build to check all new or changed links?
Yes

Is there anything else you would like us to know?
No

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+## API
+
+Attributes:
+
+| Name        | Type      | Description                                                                                                 |
+|-------------|-----------|-------------------------------------------------------------------------------------------------------------|
+| appearance  | `string`  | Changes the general view of the element; this can be **filled** or **outline**                              |
+| autofocus   | `boolean` | When true, the component will be focused when the page has finished loading                                 |
+| cols        | `boolean` | Defines the size of the element horizontally. **Default: 20**                                               |
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx:121]: is this mean to be number (as a string)?

File: examples/ui/client-capabilities/interaction/text-area.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { CodeLabel, CodeSection } from '../../documentationBase';
+import { registerComponents } from '../../rapidImports';
+
+registerComponents();
+
+export default function TextAreaDemo({ children, color }) {
+	return (<CodeSection><rapid-text-area autocapitalize="sentences" placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua">Text Area</rapid-text-area></CodeSection >)
```
Comment (patrickoneill-genesis) [inline examples/ui/client-capabilities/interaction/text-area.js:7]: Can we get more examples here?

Min length, max length, rows, cols, pre filled value, disabled etc

File: docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx (hunk 1 above)
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx:121]: If referring to cols, yeah should be number, have updated.

File: examples/ui/client-capabilities/interaction/text-area.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { CodeLabel, CodeSection } from '../../documentationBase';
+import { registerComponents } from '../../rapidImports';
+
+registerComponents();
+
+export default function TextAreaDemo({ children, color }) {
+	return (<CodeSection><rapid-text-area maxlength="20" placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua" rows="10" cols="10">Text Area</rapid-text-area></CodeSection >)
```
Comment (patrickoneill-genesis) [inline examples/ui/client-capabilities/interaction/text-area.js:7]: I meant several different text areas with different configs

Comment (Gareth-Spencer-Genesis) [reply examples/ui/client-capabilities/interaction/text-area.js:7]: I am not sure if its worth adding many different examples, the href link at the top links to the underlying component which has more examples:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea
Im not even sure it's worth adding the attributes that are there as they are all just duplicated out from the underlying component. Aren't we just duplicating information and not adding anything new?

File: docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+| resize      | `string`  | Allows the user to resize the element; this can be {"none","both","vertical","horizontal"}. **Default: none** |
+| row         | `number`  | Defines the size of the element vertically. **Default: 2**                                                  |
+| value       | `string`  | The text area content                                                                                       |
+
+
+### Examples (closing tag omitted):
+
+Attributes:
+```typescript
+<rapid-text-area placeholder="Placeholder text">
```
[the remark this answers is not in the document]
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx:138]: This code would have been in the table Example column, but I moved it underneath as the formatting is better outside of the table. When added into the table, formatting is very limited and the code is often squashed with many odd line breaks

File: examples/ui/client-capabilities/interaction/text-area.js (hunk 3 above)
Comment (patrickoneill-genesis) [reply examples/ui/client-capabilities/interaction/text-area.js:7]: I expect a lot of developers using these docs won't be front end specialists and know the difference. It's useful to show the different ways a developer can use it.

File: docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+### Parts
+
+| Part | Description |
+|---|---|
+| label | The label for the text area. |
+| control | The inpuot for the text area |
+
+### Fired Events
+
+This component doesn't emit any events.
```
[the remark this answers is not in the document]
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx:159]: Im struggling to find any documented reference to what events these things emit in general
I can see it here in the old fast documentation
https://fast.design/docs/1.x/api/fast-foundation.textarea
says it emits change, but in the latest docs its not listed.
Maybe we are supposed to just make the value observable and add the attributeChanged function to achieve the same thing?

[the remark this answers is not in the document]
Comment (patrickoneill-genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx:159]: I think it's just `change`
The source docs are in here
https://github.com/microsoft/fast/blob/master/sites/website/versioned_docs/version-1.x/components/fast-text-area.mdx

File: examples/ui/client-capabilities/interaction/text-area.js (hunk 3 above)
Comment (Gareth-Spencer-Genesis) [reply examples/ui/client-capabilities/interaction/text-area.js]: I've added two examples now

**Review by patrickoneill-genesis — APPROVED** [review]

File: examples/ui/client-capabilities/interaction/text-area.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { CodeLabel, CodeSection } from '../../documentationBase';
+import { registerComponents } from '../../rapidImports';
+
+registerComponents();
```
Comment (patrickoneill-genesis) [inline examples/ui/client-capabilities/interaction/text-area.js:4]: This needs to be fixed

**Review by patrickoneill-genesis — APPROVED** [review]

**Pull request merged**